### PR TITLE
Upgrade kubevirt go-client

### PR DESCRIPTION
**Pull request by moadqassem**
**What this PR does / why we need it**:
Upgrading k8s version to 1.16.4 and controller runtime to 0.4.0. In addition to upgrading kubevirt go client to the latest stable release.

**Special notes for your reviewer**:
At the moment there are some technical issues with kubevirt running coreos vmi, where the ignition support is somehow compromised. See this issue https://github.com/kubevirt/kubevirt/issues/3009. Once this problem is fixed we can safely upgrade the operator to the latest. 
 
**Optional Release Note**:
```release-note
None
```

### CODE REVIEW

File: pkg/admission/machines.go [hunk 1]
```diff
@@ -63,7 +63,7 @@ func (ad *admissionData) mutateMachines(ar admissionv1beta1.AdmissionReview) (*a
 		// * oldMachine has Initializers on it
```
Comment (xrstf) [inline pkg/admission/machines.go:63]: This comment isn't true anymore.

Also, can you describe why you removed the finalizer handling here?

File: pkg/apis/cluster/v1alpha1/migrations/migrations.go [hunk 2]
```diff
@@ -141,26 +142,34 @@ func MigrateMachinesv1Alpha1MachineToClusterv1Alpha1MachineIfNecessary(
 	kubeClient kubernetes.Interface,
 	providerData *cloudprovidertypes.ProviderData) error {
 
-	err := client.Get(ctx, types.NamespacedName{Name: machines.CRDName}, &apiextensionsv1beta1.CustomResourceDefinition{})
-	if err != nil {
-		if kerrors.IsNotFound(err) {
-			klog.Infof("CRD %s not present, no migration needed", machines.CRDName)
-			return nil
+	for try := 0; try < 15; try++ {
```
Comment (xrstf) [inline pkg/apis/cluster/v1alpha1/migrations/migrations.go:145]: There *has* to be a better way to wait. I'm somewhat sure that there are helpers for waiting and exponential backoffs and all that stuff.

File: pkg/admission/machines.go (hunk 1 above)
Comment (kron4eg) [reply pkg/admission/machines.go:63]: @xrstf not finalizer but initializer. Initializers has been removed from upstream k8s.io/api.

Comment (xrstf) [reply pkg/admission/machines.go:63]: Oh mea culpa. Still, the comment is now wrong, no?

Comment (moadqassem) [reply pkg/admission/machines.go:63]: Yeah that's actually my nad. the commit message is wrong :/

**Pull request merged**